runtime/runtime/src/balance_checker.rs

```diff
@@ -108,7 +108,6 @@ fn total_postponed_receipts_cost(
 
 pub(crate) fn check_balance(
     transaction_costs: &RuntimeFeesConfig,
-    initial_state: &TrieUpdate,
```

Comment:
`TrieUpdate::trie` yeah, I’m not sure about this myself.  Though since TrieUpdate is also a read interface, it makes sense to ask for a read bypassing any pending changes.  But yeah, the structs and interfaces we have are a bit of a mess.

Reply:
>But yeah, the structs and interfaces we have are a bit of a mess.

And thank *you* for cleaning them up somewhat, much appreciated!